### ISSUE TITLE
4-bit PNG file described as 8-bit

### ISSUE THREAD

**Issue by AlisterH** [ImageMagick/ImageMagick#7573]
### ImageMagick version

7.1.1-36

### Operating system

Linux

### Operating system, version and so on

tested on Nixos stable and Debian Bookworm

### Description

`identify` claims that files like this are "8-bit sRGB", but `file` says "4-bit colormap".
I believe `file` is correct, or at least closer to the truth, because after I convert to ppm* `identify` says the results are "4-bit sRGB".
This is not a new issue.

Or is there something I don't understand going on here?

*Using any version of imagemagick newer than this: https://github.com/ImageMagick/ImageMagick6/commit/7dd14de86de68f4ce88a23691ff30a91607e5894

### Steps to Reproduce

run `identify` and `file` to report on the attached image.

### Images

![test_4bit](https://github.com/user-attachments/assets/aae507c9-33d0-4935-96b2-947f77660ff1)

**Comment by snibgo** [comment]
> identify claims that files like this are "8-bit sRGB", ...

No. `identify` claims nothing about the file. Instead, it describes the image that has been read into memory.

**Comment by urban-warrior** [comment]
We get a depth of 4-bit and a meta-data tag of 4-bits.  `magick identify` without the `-verbose` option is designed to be preformant, as such, it does not do a deep analysis of the image.  Try this alternative:

```
magick image.png -format '%z' info:
```

**Comment by AlisterH** [comment]
> > identify claims that files like this are "8-bit sRGB", ...
> 
> No. `identify` claims nothing about the file. Instead, it describes the image that has been read into memory.

OK, although the man page talks about it describing image files.

> We get a depth of 4-bit and a meta-data tag of 4-bits. `magick identify` without the `-verbose` option is designed to be preformant, as such, it does not do a deep analysis of the image. Try this alternative:
> 
> ```
> magick image.png -format '%z' info:
> ```

OK, are you saying it is expected for a plain `identify` to be misleading?  If so, might it be good to make this clear to users?

**Comment by urban-warrior** [comment]
See https://imagemagick.org/Usage/basics/#identify_ping.

**Comment by AlisterH** [comment]
Would you consider adding something like this to the man page?:
"By default the -ping option is applied, which only reads in minimal data, to maximise performance.  Output may differ from that produced when reading in the full image using the +ping option."

**Comment by ilyagr** [comment]
It is quite confusing that `magick identify -help` gives no hint that `-ping` is the default, that it can be overridden with `+ping`, or why one might need to.

**Comment by ilyagr** [comment]
BTW, the docs I get are:

```
        -ping                efficiently determine image attributes
```

The website has somewhat better short docs at https://imagemagick.org/Usage/basics/#ping:

> For the "[identify](https://imagemagick.org/Usage/basics/#identify)" command. IM will try to avoid completely reading and decoding the full image file format for basic information such as the images size.

So, maybe the command-line docs will be updated soon? They still don't mention it's the default or how to disable it, though.

The docs at https://imagemagick.org/Usage/basics/#identify_ping are the best, but they are hard to find. It'd be nice if the command-line docs linked to them.
